community/server/src/test/java/org/neo4j/server/rest/transactional/ExecutionResultSerializerTest.java

```diff
@@ -82,14 +83,30 @@
 import static org.neo4j.test.mocking.GraphMock.relationship;
 import static org.neo4j.test.mocking.Properties.properties;
 
-public class ExecutionResultSerializerTest
+public class ExecutionResultSerializerTest extends TxStateCheckerTestSupport
```

Comment:
The id gets returned I guess because it is the only thing that we _can_ return. The `deleted` flag could be checked in order for the client to not get exceptions when trying to lookup properties of the deleted entities. Also, it differentiates the returned entities from non-deleted, property-less dittos. Returning the node _without_ properties (if it had any) would be bad. Returning _with_ properties is not possible.